### PR TITLE
allof, oneof, discriminator support 

### PR DESCRIPTION
**Pull request by KlimenkoAV**

### CODE REVIEW

File: src/swagger/OpenAPIService.ts [hunk 1]
```diff
@@ -233,9 +242,38 @@ export class OpenAPIService {
         return refs;
     }
 
+    /**
+     * @description Gets refs from allof schema only one level down
+     */
+    private getRefsByAllOf(
```
Comment (aleshchev) [inline src/swagger/OpenAPIService.ts:248]: looks like totally copy paste from getRefsByObject. Can you combine methods

File: src/swagger/OpenAPIService.ts [hunk 2]
```diff
@@ -203,6 +206,12 @@ export class OpenAPIService {
                 const expanded = this.expandRefs(refsFromObject, refKeys);
                 collectedRefs.push(...expanded);
             }
+
```
Comment (aleshchev) [inline src/swagger/OpenAPIService.ts:209]: combine with isObject case

File: src/swagger/OpenAPIService.ts (hunk 1 above)
Comment (KlimenkoAV) [reply src/swagger/OpenAPIService.ts]: done

File: src/swagger/OpenAPIService.ts (hunk 2 above)
Comment (KlimenkoAV) [reply src/swagger/OpenAPIService.ts]: done

**Comment by akhromets** [comment]
Please increase the package version.

File: src/generators/utils/modelGuard.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { IExtendedInterfaceModel, InterfaceModel } from '../../models/InterfaceModel';
+
```
Comment (aleshchev) [inline src/generators/utils/modelGuard.ts:2]: Why is it not in TypeGuard service ?

**Review by aleshchev — APPROVED** [review]

**Pull request merged**